src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -275,6 +267,53 @@ private static boolean changeIsLargerThanThreshold(
             ||  deltaBwe < -1 * previousBwe * Config.bweChangeThresholdPct() / 100;
     }
 
+    static class EndpointMultiRank
+    {
+        final int speakerRank;
+        final EndpointConstraints endpointConstraints;
+        final AbstractEndpoint endpoint;
+
+        EndpointMultiRank(int speakerRank, EndpointConstraints endpointConstraints, AbstractEndpoint endpoint)
+        {
+            this.speakerRank = speakerRank;
+            this.endpointConstraints = endpointConstraints;
+            this.endpoint = endpoint;
+        }
+    }
+
+    static class EndpointMultiRanker
+        implements Comparator<EndpointMultiRank>
+    {
+
+        @Override
+        public int compare(EndpointMultiRank o1, EndpointMultiRank o2)
+        {
+            // An endpoint that has higher priority/rank will be allocated
+            // bandwidth prior to other endpoints with lower priority/rank
+            // (see the allocate method bellow)
+            //
+            // Once the endpoints are ranked, the bandwidth allocation algorithm
+            // loops over the endpoints multiple times, improving their target
+            // bitrate at ever step, until no further improvement is possible.
+            //
+            // In this multi-rank implementation, endpoints that have 720p ideal
+            // height will be given bandwidth before endpoints that have 180p
+            // ideal height. If two endpoints have the same ideal height, then
+            // we look at their speech rank (whoever spoke last has is ranked
+            // higher).
+
+            int idealHeightDiff = o1.endpointConstraints.getIdealHeight() - o2.endpointConstraints.getIdealHeight();
```

Comment:
Making sure I understand it right: this ranking is specifically for bandwidth allocation, which is why we prioritize first by ideal height, and then by speaker rank?

Reply:
Yes, which means that we allocate bandwidth first for "selected" endpoints (720p ideal resolution), starting with the most recent active speaker in that group, then 180p again starting with the most recent active speaker in that group and then everything else.